### PR TITLE
Theme update

### PR DESCRIPTION
**Pull request by erogluorhan**
This is now a working `sphinx-pythia-theme` update with the followings:

- `sphinx_pythia_theme/layout.html` updated to keep our website's layout/style almost 100% unchanged by using the layout/style changes in the parent `sphinx_pythia_theme`
- Keep using our `sphinx_pythia_theme/footer.html` (though with some updates in it) to create our custom Pythia footer using the parent's `footer_start` only
- `src/scss/sphinx-pythia-theme.scss` updated (a lot of `!important` used) to keep Pythia's custom colors and styles working with the changes in the parent themes
- A lot of layout / style related config and html_theme_options changes here and there
- `sphinx-book-theme>=1.0.0`
- Python=3.11
- Sphinx=7.2.6 tested
- Infratstructural housekeeping (e.g. CI workflows, etc.)
- Emails, copyright year, etc. updated

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -7,61 +7,73 @@ on:
...
   jupyterbook:
-    name: Build Jupyter Book
+    name: Build Jupyter Book and Sphinx Docs
     runs-on: ubuntu-latest
+
     steps:
     - uses: actions/checkout@v2
+
+    - name: Cancel previous runs
+      uses: styfle/cancel-workflow-action@0.12.0
```
[the remark this answers is not in the document]
Comment (erogluorhan) [reply .github/workflows/ci.yaml:21]: Thanks a lot; wasn't aware of this key. Just went over its docs and found it a bit more complicated than what is currently here (e.g. defining groups, specifying if concurrency for PR numbers, etc.), though please feel free to push the change if you'd like that better (and maybe find it more useful than the one here). 

If I were to implement it, should it look something like this:

```
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -7,61 +7,73 @@ on:
...
         with:
-          python-version: 3.8
-      - name: Install build dependecies
+          python-version: 3.11
+
+      - name: Install build dependencies
         run: pip install setuptools setuptools-scm wheel
+
       - name: Build package
         run: python setup.py sdist bdist_wheel
```
[the remark this answers is not in the document]
Comment (erogluorhan) [reply .github/workflows/ci.yaml:75]: Good catch! Even though this existing one should probably keep working, I will look into replacing this with `build` (hadn't use it before). Please feel free to push if you have that close at hand

File: docs/about.rst [hunk 3]
```diff
@@ -31,7 +31,7 @@ Additionally, the
 (``external_links``) option from the PyData Sphinx Theme still works, and these links
 will be displayed after any links specified by the ``navbar_links`` option.
 
-.. tabbed:: Sphinx
+.. tab-set-code:: Sphinx
```
[the remark this answers is not in the document]
Comment (erogluorhan) [reply docs/about.rst:34]: Great catch, removed.

File: .github/workflows/ci.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (erogluorhan) [reply .github/workflows/ci.yaml]: I think I have incorporated this now. Please have a look

**Comment by erogluorhan** [comment]
An FYI: The Pythia portal nightly build with this updated theme [built successfully](https://github.com/ProjectPythia/projectpythia.github.io/actions/runs/7706845158/job/21003043090); however, we will need to inspect its [artifact](https://github.com/ProjectPythia/projectpythia.github.io/actions/runs/7706845158/artifacts/1204623717) and maybe perform additional tests to investigate the results of the theme is good for actual deployment.

**Review by jukent — APPROVED** [review]
@erogluorhan This looks pretty reasonable to me and I'm glad that it is working. However it's possible there are things I don't understand well enough to catch. Let me know if you want to do a walk through of the changes. 

Also I'm wondering if there are any lessons you learned while working on this that could go into some internal documentation for ease of future updates.

**Comment by erogluorhan** [comment]
Hey @jukent , thanks for that! Yes, I can give you a walk through of the current state anytime. Even though I marked this as review-ready, it was more for bringing it to attention of everyone in the loop, and get some feedback/help from the more experienced (since I am not at all experienced on this work). 

The current state of it seems good about that we were able to get Sphinx, theme, and dependency packages up & running with their unpinned versions. However, there is still work to do on getting this new version of the theme to reflect all the features listed [here](https://sphinx-pythia-theme.readthedocs.io/en/latest/about.html). Currently, I am looking into `layout.html` to display the banner page layout, hide the right ToC, etc. in Pythia portal's homepage properly as before.

**Comment by erogluorhan** [comment]
FYI: [This](https://github.com/ProjectPythia/projectpythia.github.io/actions/runs/7806163431/artifacts/1225049705) is where I am as of now. You can download it and check out `_build/html/index.html` locally.

File: .github/workflows/ci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (agoose77) [reply .github/workflows/ci.yaml:21]: Sorry, I didn't see this outstanding question. Yes, the concurrency group just needs to be non-unique for the same PR e.g.
```
group: 'docs-${{ github.head_ref || github.run_id }}'
```

This ensures that if the run doesn't have a `head_ref` (non-PRs), then it falls back to a unique identifier.

**Comment by jukent** [comment]
I am getting an error trying to build the Foundations book because the Cartopy chapter starts with an illegal character. @dcamron do you get a similar error? It might not be related to the theme.

**Review by jukent — CHANGES_REQUESTED** [review]
Screenshot 2024-03-22 at 11 26 54 AM

The navbar menu links' color is set to `--pst-color-text-muted`. I think this is too light. On the site now, they're a dark gray that turns teal when hovered over.

The underlining of the link when hovering over it is also new, but I think helpful.

Great work on this! I know it was really a big push.

**Comment by jukent** [comment]
Steps taken to build.

From Portal or Foundations repository:

`git fetch --all`
`git branch -r`
` git checkout -b theme upstreme/try_updated_theme`
` conda env create -f environment.yml`

Then from the sphinx-pythia-theme repository:

`conda activate pythia-book-dev` (for foundations) or `conda activate pythia` (for portal)
 `pip install -r requirements.txt`
`python -m pip install . --no-deps`

Then in Portal
`cd portal`
`make html`
`open _build/html/index.html`

Or in Foundations
`jupyter-book build .`
open _build/html/index.html`

**Comment by erogluorhan** [comment]
image

How does it look now, @jukent? (screenshot is from my local but I can push it if it is liked)

**Comment by jukent** [comment]
Yeah that looks great! If you push the commit I'll re-test my build.

**Review by jukent — APPROVED** [review]
Thanks Orhan for all of the hard work that went into this. These changes seem sensible and we've had several chances to look at the Html together. Thanks for updating the CSS to make the foundations navbar more legible. 

LGTM! I think this issue is holding up another [PR](https://github.com/ProjectPythia/pythia-foundations/pull/452) and plenty of other book dev work. Thanks again!

**Comment by ktyle** [comment]
@erogluorhan and I are meeting to chat at 11AM CDT today (@jukent feel free to join as well!), but I agree that we're on the right track and it will be great to get this merged soon since it will resolve a lot of existing issues. Some things I notice, cookbook-related at least, include:

1. Some cookbooks have Jupyter notebooks whose output does not properly fit in the main (center) panel. In the cases I've tested so far, they have all involved Holoviz/Bokeh-based visualizations.
2. At least on my MacBook Air, the left and right panels are too wide, yet the text does not fully occupy the panel.

File: .github/workflows/ci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (erogluorhan) [reply .github/workflows/ci.yaml:21]: No worries! FYI: The theme update is working now, and its proper functioning (with a few non-vital issues) has been confirmed through tests by a few team members. It is a bit time sensitive before our upcoming community event; hence, I'd like to get it merged and go forward to be used by our portal. 

I've created a new issue though to address this suggestion.

Thanks very much for all your help and review!

**Comment by erogluorhan** [comment]
Hey @agoose77 please see this [discussion about the theme update](https://github.com/ProjectPythia/projectpythia.github.io/discussions/420). Thanks for everything!

**Pull request merged**